Simplified/AxisService/NYPLAxisService.swift

```diff
@@ -75,16 +129,194 @@ class NYPLAxisService: NSObject {
     self.baseURL = AxisHelper.baseURL.appendingPathComponent(isbn)
     self.book = book
     self.deviceInfoProvider = deviceInfoProvider
+    self.packagePathProvier = NYPLAxisPackagePathPrefixProvider()
+    self.dispatchGroup = DispatchGroup()
     self.dedicatedWriteURL = fileURL
       .deletingLastPathComponent()
       .appendingPathComponent(book.identifier.sha256())
   }
   
   /// Fulfill AxisNow license. Notifies NYPLBookDownloadBroadcasting upon completion or failure.
+  
   /// - Parameters:
   ///   - downloadTask: downloadTask download task
   @objc func fulfillAxisLicense(downloadTask: URLSessionDownloadTask) {
+    DispatchQueue.global(qos: .utility).async {
+      self.downloadLicense(forBook: self.book)
+      self.downloadEncryption(forBook: self.book)
+      self.downloadContainer(forBook: self.book)
```

Comment:
If one of the processes fails, we should not go to the next process since the downloaded book will be useless. When we encounter failure, we should delete all the downloaded files and stop all subsequent calls.